File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.java
import java.util.HashMap;
import java.util.HashSet;

class Solution {
	public int[] solution(String[] id_list, String[] report, int k) {
		int[] answer = new int[id_list.length];
		HashMap<String, Integer> idMap = new HashMap<>() {
			{
				int i = 0;
				for (String key : id_list)
					put(key, i++);
			}
		};
		HashMap<String, HashSet<String>> map1 = new HashMap<>() {
			{
				for (String key : id_list)
					put(key, new HashSet<>());
			}
		};
		for (String str : report) {
			String[] origin = str.split(" ");
			map1.get(origin[1]).add(origin[0]);
		}
		for (HashSet<String> arr : map1.values())
			for (String s : arr)
				if (arr.size() >= k) {
					answer[idMap.get(s)]++;
				}
		return answer;
	}
}

File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    answer = [0] * len(id_list)
    id_dict, report_dict = {}, {}
    idx = 0
    for id in id_list:
        id_dict[id] = idx
        report_dict[id] = []
        idx += 1
        
    report_set = set(report)

    for r in report_set:
        reporter, reportee = r.split(" ")
        report_dict[reportee].append(reporter)

    for v_list in report_dict.values():
        if len(v_list) >= k:
            for v in v_list:
                answer[id_dict[v]] += 1
    return answer